debezium-core/src/main/java/io/debezium/pipeline/ChangeEventSourceCoordinator.java

```diff
@@ -107,7 +108,7 @@ public ChangeEventSourceCoordinator(TaskOffsetContext<P, O> previousOffsetContex
                 P partition = null;
                 O previousOffset = null;
 
-                Map<P, SnapshotResult<O>> partitionState = new HashMap<>();
+                Map<P, SnapshotResult<O>> partitionState = new LinkedHashMap<>();
```

Comment:
Would it make sense to leave it as `HashMap` and instead of iterating over the `partitionState`, iterate over `previousOffsetContext.getOffsets()`? This way, the offset context remains the source of truth both about the partitions to be processed and the order of processing.